From a25c7f3572eb7c904356df561bd9f850a83ed473 Mon Sep 17 00:00:00 2001
From: Jerry Cao <jerry.cao@amlogic.com>
Date: Fri, 14 Mar 2014 13:06:07 -0700
Subject: [PATCH 3737/5965] Revert "pinctrl: protect pinctrl_list add"

Temporarily revert 3.10 pinctrl driver fix before switch back to
standard 3.10 pinctrl driver.

This reverts commit 1fc42b84b45d4a58641821c3c0b601565523ad7d.
---
 drivers/pinctrl/core.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/pinctrl/core.c b/drivers/pinctrl/core.c
index f0f5692984e0..1298357b614f 100755
--- a/drivers/pinctrl/core.c
+++ b/drivers/pinctrl/core.c
@@ -705,9 +705,7 @@ static struct pinctrl *create_pinctrl(struct device *dev)
 	}
 	mutex_unlock(&pinctrl_maps_mutex);
 	/* Add the pinctrl handle to the global list */
-	mutex_lock(&pinctrl_list_mutex);
 	list_add_tail(&p->node, &pinctrl_list);
-	mutex_unlock(&pinctrl_list_mutex);
 
 	return p;
 }
-- 
2.19.0

